tests/lib/rules/valid-typeof.js

```diff
@@ -144,17 +149,17 @@ ruleTester.run("valid-typeof", rule, {
         {
             code: "typeof foo === undefined",
             options: [{ requireStringLiterals: true }],
-            errors: [{ messageId: "notString", type: "Identifier" }]
+            errors: [{ messageId: "invalidValue", type: "Identifier" }]
         },
```

Comment:
I think that `messageId: "notString"` makes more sense here. If `"requireStringLiterals": true` is set, and `sibling` isn't a string literal (or simple template literal or another `typeof`), then its value shouldn't matter because it must be a string literal anyway, so that check should be first. We could just reorder conditions in the if-else-if chain to get the following behavior:

```js
/* eslint valid-typeof: ["error", { "requireStringLiterals": true }] */

typeof foo === undefined; //  error  Typeof comparisons should be to string literals  valid-typeof
```

```js
/* eslint valid-typeof: ["error"] */

typeof foo  === undefined; //  error  Invalid typeof comparison value  valid-typeof

```